docs/docs/sample-scripts/entra/find-obsolete-m365-groups/index.mdx

```diff
@@ -0,0 +1,282 @@
+    if ($TestPath -ne $true) {
+      New-Item -ItemType directory -Path $Script:Path | Out-Null
+      Write-Host "Will create file in $($Script:Path): M365GroupsReport-{current date}.csv" -ForegroundColor Yellow
+    }
+    else {
+      Write-Host "Following report file will be created in $($Script:Path): 'M365GroupsReport-$($tStamp).csv'."
+      Write-Host "`nAll data will be exported to $($Script:Path): M365GroupsReport-$($tStamp).csv." -ForegroundColor Blue
+      Write-Host "Do not edit this file during the scan." -ForegroundColor Blue
+    }
+    $Script:ReportPath = "$($Script:Path)/M365GroupsReport-$($tStamp).csv"
+  }
+
+  function Get-AllM365Groups {
+    $groups = m365 entra m365group list --includeSiteUrl | ConvertFrom-Json
+    $Script:Groups = $groups | Where-Object { $null -ne $_.siteUrl }
+  }
+
+  function Get-AllGuestUsers {
+    $Script:Guests = m365 entra user list --type Guest | ConvertFrom-Json
+  }
+
+  function Get-AllTeamSites {
+    $Script:TeamSites = m365 spo site list --type TeamSite | ConvertFrom-Json
+  }
+
+  function Start-GroupInsightsTests {
+    Write-Host "Checking $($Script:Groups.Count) groups for activity"
+
+    $Script:Groups | ForEach-Object {
+      $groupInfo = [GroupInfo]::new()
+      $groupInfo.Reference = $_
+      $groupInfo.Membership = @{Owners = 0; Members = 0; Guests = 0 }
+      $groupInfo.TestStatus = "🟢 OK"
+
+      Write-Host "☀︎ $($groupInfo.Reference.displayName)"
+
+      # Tests
+      Test-GroupMembership -Group $groupInfo
+      Test-SharePointActivity -Group $groupInfo
+      Test-ConversationActivity -Group $groupInfo
+
+      # Report
+      New-Report -Group $groupInfo
+    }
+    
+    #Give feedback to user
+    Write-Host "`n-------------------------------------------------------------------"
+    Write-Host "`SUMMARY" -ForegroundColor DarkGreen
+    Write-Host "`-------------------------------------------------------------------"
+    Write-Host "`n👉 Found $($Global:ObsoleteGroups.Count) group$($Global:ObsoleteGroups.Count -gt 1 ? 's' : '') with possibly low activity."
+    Write-Host "`   Please review the report: " -NoNewline
+    Write-Host "$($Script:ReportPath)" -ForegroundColor DarkBlue
+  }
+
+  function Test-GroupMembership {
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory = $true)] [GroupInfo] $Group
+    )
+
+    # Original lists
+    $users = m365 entra m365group user list --groupId $Group.Reference.id | ConvertFrom-Json
+    $owners = m365 entra m365group user list --groupId $Group.Reference.id --role Owner | ConvertFrom-Json
+    $members = m365 entra m365group user list --groupId $Group.Reference.id --role Member | ConvertFrom-Json
+    
+    # Consider guests as users that also match the $Script:Guests array
+    $guests = $users | Where-Object { $_.id -in $Script:Guests.id }
+
+    # Modify the $members list to only contain users that are not in the $owners list
+    $members = Compare-Object $members $owners -PassThru
```

Comment:
This piece of code is still failing for all company groups which have no owners and no members.

Reply:
You're absolutely right. This would be the case, when every group member has left the company. In this case, any comparison would be obsolete.

Reply:
As mentioned in the previous review. When you have an all company group, it doesn't have any owners or members.

Reply:
@milanholemans That's why I've implemented this on line 170:
```
if($null -ne $owners -and $null -ne $members) {
  $members = Compare-Object $members $owners -PassThru
}
```
Doesn't it show up in the code?

Reply:
The last time I reviewed it wasn't there. I'll recheck in my next review.